src/main/java/com/google/netpcapanalysis/dao/PCAPParserDaoImpl.java

```diff
@@ -0,0 +1,166 @@
+  public void parseRaw() throws IOException {
+    final InputStream stream = PCAPParserDaoImpl.class.getClassLoader().getResourceAsStream(FILENAME);
+    final Pcap pcap = Pcap.openStream(stream);
+
+    pcap.loop(new PacketHandler() {
+      @Override
+      public boolean nextPacket(final Packet packet) throws IOException {
+      if(packet.hasProtocol(Protocol.IPv4)) {
+        IPPacket ip = (IPPacket) packet.getPacket(Protocol.IPv4);
+
+        String srcip = ip.getSourceIP();
+        String dstip = ip.getDestinationIP();
+
+        Buffer buffer = ip.getPayload();
+        int size = buffer.getReadableBytes();
+
+        String protocol = getProtocol(ip); 
+
+        if (first) {
+            MYIP = srcip;
+            first = false;
+        }
+        
+        // PCAPdata takes in (source, destination, domain, location, protocol, size, flagged, frequency) 
+        PCAPdata rawPCAP = new PCAPdata(srcip, dstip, "", "", protocol, size, false, 1);
+        allPCAP.add(rawPCAP);
+      }
+      return true;
+    }
+   });
+   pcap.close();
+  } 
+
+  /* Determine the protocol of IPv4 packet; if unknown, then returns "IPv4" as protocol.
+    * Checks for the following protocols — ICMP, IGMP, TCP, UDP, SCTP, SIP, SDP, ETHERNET_II, SLL, IPv4, RTP —
+    * which are the only protocols supported by pkts library. */
+  private String getProtocol(IPPacket packet) throws IOException {
+    String protocol = "IPv4";
+    // If packet has UDP or TCP protocol, fetch the port numbers (source/destination).
+    if (packet.hasProtocol(Protocol.UDP)) {
+      protocol = "UDP";
+      UDPPacket udpPacket = (UDPPacket) packet.getPacket(Protocol.UDP);
+      int dstport = udpPacket.getDestinationPort();
+      int srcport = udpPacket.getSourcePort();
+    }
+    else if (packet.hasProtocol(Protocol.TCP)) {
+      protocol = "TCP";
+      TCPPacket tcpPacket = (TCPPacket) packet.getPacket(Protocol.TCP);
+      int dstport = tcpPacket.getDestinationPort();
+      int srcport = tcpPacket.getSourcePort();
+    } 
+    else if (packet.hasProtocol(Protocol.ICMP)) {
+      protocol = "ICMP";
+    }
+    else if (packet.hasProtocol(Protocol.IGMP)) {
+      protocol = "IGMP";
+    }
+    else if (packet.hasProtocol(Protocol.SCTP)) {
+      protocol = "SCTP";
+    }
+    else if (packet.hasProtocol(Protocol.SIP)) {
+      protocol = "SIP";
+    }
+    else if (packet.hasProtocol(Protocol.SDP)) {
+      protocol = "SDP";
+    }
+    else if (packet.hasProtocol(Protocol.ETHERNET_II)) {
+      protocol = "ETHERNET_II";
+    }
+    else if (packet.hasProtocol(Protocol.SLL)) {
+      protocol = "SLL";
+    }
+    return protocol;
+  }
+ 
+  /* Removes duplicated data and increments corresponding frequencies */
+  public void processData(){
+    for (PCAPdata packet : allPCAP) {
+      String OUTIP = "";
+      String srcip = packet.source;
+      String dstip = packet.destination;
+
+      if (srcip == MYIP) {
```

Comment:
Since MYIP is the local address and OUTIP is the "outside" machine, in the following packets the order could be flipped where (MYIP, OUTIP) are outgoing packets and (OUTIP, MYIP) are incoming packets; we're using OUTIP as the key in the hashmap since that's always unique.

Reply:
Thanks for the clarification